packages/vue-components/src/Dropdown.vue

```diff
@@ -123,15 +124,23 @@ export default {
     },
     hideDropdownMenu() {
       this.show = false;
-      $(this.$refs.dropdown).findChildren('ul').each(ul => ul.classList.toggle('show', false));
+      this.nodeList(this.$refs.dropdown).findChildren('ul').each(ul => ul.classList.toggle('show', false));
```

Comment:
Can I check if you are suggesting me to use Webpack to "stub" the browser variables (e.g. `window`)?

Just to give more context for my workaround, the problem I faced was having `window` and `document` in `NodeList`. Webpack will raise errors about the undefined variables `window` and `document` during compilation. 

Even though these variables are not supposed defined as we are not in a browser environment, a straightforward resolution would be to define these undefined variables so that Webpack would not raise errors about it. 

However, by defining the `window` variable, we run into another issue where Vue would wrongly detect the environment as the browser environment and would try to execute some browser-related operations (e.g. reading `userAgent` property of `window`). Thus, I thought that defining `window` variable as an empty object would not be suitable workaround.

![Screen Shot 2021-04-11 at 3 18 31 PM](https://user-images.githubusercontent.com/37920115/114295657-2f796e00-9ad9-11eb-9e7a-c167b7e1f17b.png)

So the workaround that I decided on is to "define" browser variables such as `window` by passing it as parameters for use in `NodeList`, where `window` would only be passed in to create the `NodeList` instance during the mounting lifecycle in Vue, so Vue would not falsely detect the environment to be the browser environment. 

Are you suggesting that there is perhaps an easier workaround by using Webpack?

Reply:
> Can I check if you are suggesting me to use Webpack to "stub" the browser variables (e.g. window)?

Yup, this should be a rather common use case; Stubbing it for each individual usage might be a little tedious for future development.

> However, by defining the window variable, we run into another issue where Vue would wrongly detect the environment as the browser environment and would try to execute some browser-related operations (e.g. reading userAgent property of window). Thus, I thought that defining window variable as an empty object would not be suitable workaround.

would defining it as `undefined` work? its probably safer as well, since things might get run on an empty object.

Reply:
Thanks for the suggestion! Defining it as `undefined` would work, I just have to add some guard clauses whenever we try to read the properties of these `undefined` global browser variables (e.g. have to add a check for `window` before we read `window.Symbol`). The outcome is essentially the same as my original implementation, but a lot more succinct and straightforward :-)

Reply:
Sorry @wxwxwxwx9, realised I misread your reply here earlier after observing the current changes.

It should essentially just be a 2-line solution (setting up `definePlugin`) to allow webpack to compile the bundle.
i.e. the `window && ...` checks should not even be needed as it indicates ssr is hitting some part of the code it shouldn't be.

Reply:
@ang-zeyu I've tested it out and I think the problems stems from my usage of `requireFromString`, which does module compilation `m._compile`. This is the part which hits the code that you mentioned (and thus guard clauses are needed). 

In view of this, I tried to look for an alternative to `requireFromString` (perhaps require the module directly?) but I don't think there's a clear alternative. I'm thinking that we still have to use `requireFromString` nonetheless because of how `memory-fs` works; its API only provides `readFileSync` so we can only get the string version of the module from webpack watcher. 

Do you know if there's other ways around this? Or is it alright if we leave the guard clauses as they are currently?